### ISSUE TITLE
UnitaryHack Bounty - Quantum Machine Learning Tutorial: Music Genre Classification using HierarQcal

### ISSUE THREAD

**Issue by matt-lourens** [matt-lourens/hierarqcal#48]
**Issue**: Create a quantum machine learning tutorial in a Jupyter notebook using the Python package **HierarQcal** for classifying music genres.

## Overview
This tutorial will guide users on using HierarQcal to create quantum circuit models for classification tasks. The goal is to provide a quantum machine learning template, where a user can obtain baseline model performance for a classification task. While focusing on the GTZAN dataset's country vs. rock genre pair, the tutorial should emphasize a robust pipeline for both classical and quantum aspects of QML.

#### Resources:

- **Paper**: The methods section in the hierarqcal paper serves as a blueprint for constructing a machine learning pipeline. The Architectural Impact section will also help guide model development and provide something to compare against.

- **qml_example.py**: A code snippet is provided as an MVP for this tutorial, and should be expanded into a comprehensive, well-explained template.

- **Data**: Use the [GTZAN dataset](https://www.kaggle.com/datasets/andradaolteanu/gtzan-dataset-music-genre-classification?resource=download) to distinguish between country and rock songs as done in the paper. Use the 30 second window data with PCA(8 components) encoded on 8 qubits with AngleEmbedding. If alternative preprocessing yields better results, feel free to implement it.

## Key Notes
- Ensure the tutorial is adaptable to other classification tasks despite the specific example used.
- Suggested sections (feel free to modify as needed):
  1. **Data Loading**
  2. **Data Cleaning and Preprocessing**
  3. **Model Creation (with HierarQcal)**
  4. **Training Pipeline**
  5. **Evaluation and Visualization**
- Implement cross-validation for a comprehensive model evaluation.
- Feel free to re-use images/graphs from the paper to include in the tutorial.

## Requirements
- **Code Base**: Utilize the provided code, and expand or improve upon it as needed. The code includes essential parts for getting a model that works.
- **Explanation**: Provide detailed explanations for each code section to clarify the quantum circuit architecture and machine learning pipeline.
- **Visualization**: Incorporate plots that illustrate model architecture, training loss curves, and overall performance.

## References
- Lourens et al. (2023). **Hierarchical Quantum Circuit Representations for Neural Architecture Search**.
  - [hierarqcal_paper.pdf](https://github.com/matt-lourens/hierarqcal/files/15216132/s41534-023-00747-z.pdf)
- `qml_example.py`
  -  https://github.com/matt-lourens/hierarqcal/blob/develop/tests/qml_example_mvp.py

**Comment by Gopal-Dahale** [comment]
@matt-lourens Created a PR on QML for music genre classification. I will need your help on the different QCNN architectures used in the paper to extend the tutorial and benchmark them.
